google/resource_compute_vpn_gateway.go

```diff
@@ -60,42 +82,68 @@ func resourceComputeVpnGateway() *schema.Resource {
 
 func resourceComputeVpnGatewayCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	network, err := ParseNetworkFieldValue(d.Get("network").(string), d, config)
+
+	project, err := getProject(d, config)
 	if err != nil {
 		return err
 	}
 
-	region, err := getRegion(d, config)
+	descriptionProp, err := expandComputeVpnGatewayDescription(d.Get("description"), d, config)
 	if err != nil {
 		return err
 	}
-
-	project, err := getProject(d, config)
+	nameProp, err := expandComputeVpnGatewayName(d.Get("name"), d, config)
+	if err != nil {
+		return err
+	}
+	networkProp, err := expandComputeVpnGatewayNetwork(d.Get("network"), d, config)
+	if err != nil {
+		return err
+	}
+	regionProp, err := expandComputeVpnGatewayRegion(d.Get("region"), d, config)
 	if err != nil {
 		return err
 	}
 
-	name := d.Get("name").(string)
-
-	vpnGatewaysService := compute.NewTargetVpnGatewaysService(config.clientCompute)
+	obj := map[string]interface{}{
+		"description": descriptionProp,
+		"name":        nameProp,
+		"network":     networkProp,
+		"region":      regionProp,
+	}
 
-	vpnGateway := &compute.TargetVpnGateway{
-		Name:    name,
-		Network: network.RelativeLink(),
+	url, err := replaceVars(d, config, "https://www.googleapis.com/compute/v1/projects/{{project}}/regions/{{region}}/targetVpnGateways")
+	if err != nil {
+		return err
 	}
 
-	if v, ok := d.GetOk("description"); ok {
-		vpnGateway.Description = v.(string)
+	log.Printf("[DEBUG] Creating new VpnGateway: %#v", obj)
+	res, err := Post(config, url, obj)
+	if err != nil {
+		return fmt.Errorf("Error creating VpnGateway: %s", err)
 	}
 
-	op, err := vpnGatewaysService.Insert(project, region, vpnGateway).Do()
+	// Store the ID now
+	id, err := replaceVars(d, config, "{{name}}")
 	if err != nil {
-		return fmt.Errorf("Error Inserting VPN Gateway %s into network %s: %s", name, network.Name, err)
+		return fmt.Errorf("Error constructing id: %s", err)
 	}
+	d.SetId(id)
 
-	err = computeOperationWait(config.clientCompute, op, project, "Inserting VPN Gateway")
+	op := &compute.Operation{}
+	err = Convert(res, op)
 	if err != nil {
-		return fmt.Errorf("Error Waiting to Insert VPN Gateway %s into network %s: %s", name, network.Name, err)
```

Comment:
This error is a lot better - I see that it'd be a pain to autogenerate it, but how close can we get?

Reply:
Done!